cdk/domino_cdk/config/__init__.py

```diff
@@ -0,0 +1,115 @@
+from semantic_version import Version
+
+from domino_cdk import __version__
+from domino_cdk.config.base import DominoCDKConfig
+
+
+def config_loader(c: dict):
+    schema = Version(c.pop("schema", "0.0.0")).truncate()
+    if schema == Version("0.0.1"):
+        return DominoCDKConfig.from_0_0_1(c)
+    elif schema == Version("0.0.0"):
+        return DominoCDKConfig.from_0_0_0(c)
```

Comment:
would definitely prefer some sort of schema registration system so we don't need a lot of if/else (and could validate that the schema itself is a valid identifier)

Reply:
It's less involved than what you were demonstrating in slack, but I made it dynamic and added some related schema version validation tests in the unit tests.